### PR TITLE
Improving test coverage

### PR DESCRIPTION
**Pull request by HamsaV**
## Purpose
Improving test coverage

## Goals
Adding test cases
adding encoding details to the pom as properties

## Security checks
 - Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes
 - Ran FindSecurityBugs plugin and verified report? yes
 - Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes

### CODE REVIEW

File: component/src/test/java/org/wso2/extension/siddhi/io/mqtt/sink/MqttPublishTestCase.java [hunk 1]
```diff
@@ -0,0 +1,76 @@
+package org.wso2.extension.siddhi.io.mqtt.sink;
```
Comment (tishan89) [inline component/src/test/java/org/wso2/extension/siddhi/io/mqtt/sink/MqttPublishTestCase.java]: licence please

File: component/src/test/java/org/wso2/extension/siddhi/io/mqtt/sink/MqttSinkTestCase.java [hunk 2]
```diff
@@ -77,27 +78,24 @@ public void mqttPublishSingleEvent() {
...
-                        "@map(type='xml'))" +
-                        "Define stream BarStream (symbol string, price float, volume long);" +
-                        "from FooStream select symbol, price, volume insert into BarStream;");
+                "define stream FooStream (symbol string, price float, volume long); " + "@info(name = 'query1') "
+                        + "@sink(type='mqtt', url= 'tcp://localhost:1883', "
+                        + "topic='mqtt_publish_single_event',username='mqtt-user', "
+                        + "password='mqtt-password', clean.session='true', message.retain='false', "
+                        + "quality.of.service= '1', keep.alive= '60'," + "@map(type='xml'))"
+                        + "Define stream BarStream (symbol string, price float, volume long);"
+                        + "from FooStream select symbol, price, volume insert into BarStream;");
```
Comment (tishan89) [inline component/src/test/java/org/wso2/extension/siddhi/io/mqtt/sink/MqttSinkTestCase.java]: Aren't we asserting barStream

File: component/src/test/java/org/wso2/extension/siddhi/io/mqtt/sink/MqttSinkTestCase.java [hunk 3]
```diff
@@ -390,4 +362,48 @@ public void mqttPublishWithoutBrokerCredentials() {
...
+            siddhiAppRuntime.restoreLastRevision();
+            fooStream.send(new Object[] { "IBM", 75.6f, 100L });
+            Thread.sleep(500);
+
+        } catch (InterruptedException e) {
+            AssertJUnit.fail("Thread sleep was  interrupted");
+        }
+        count = mqttTestClient.getCount();
+        eventArrived = mqttTestClient.getEventArrived();
+        AssertJUnit.assertEquals(4, count);
```
Comment (tishan89) [inline component/src/test/java/org/wso2/extension/siddhi/io/mqtt/sink/MqttSinkTestCase.java]: Can we also assert for the event payload

File: component/src/test/java/org/wso2/extension/siddhi/io/mqtt/source/MqttSourceTestCase.java [hunk 4]
```diff
@@ -277,23 +266,20 @@ public void receive(Event[] events) {
...
+                + "define stream FooStream (symbol string, price float, volume long); " + "@info(name = 'query1') "
+                + "@sink(type='mqtt', url= 'tcp://localhost:1883', "
+                + "topic='mqtt_receive_without_clean_session', clean.session='true', message.retain='false', "
+                + "quality.of.service= '1',keep.alive= '60'," + "@map(type='xml'))"
+                + "Define stream BarStream (symbol string, price float, volume long);"
+                + "from FooStream select symbol, price, volume insert into BarStream;");
         InputHandler fooStream = siddhiAppRuntime.getInputHandler("FooStream");
         siddhiAppRuntime.start();
 
         ArrayList<Event> arrayList = new ArrayList<Event>();
```
Comment (tishan89) [inline component/src/test/java/org/wso2/extension/siddhi/io/mqtt/source/MqttSourceTestCase.java]: Declaration should be a List

Comment (tishan89) [reply component/src/test/java/org/wso2/extension/siddhi/io/mqtt/source/MqttSourceTestCase.java]: Change in all places

File: component/src/test/java/org/wso2/extension/siddhi/io/mqtt/source/MqttSourceTestCase.java [hunk 5]
```diff
@@ -612,4 +572,70 @@ public void receive(Event[] events) {
...
+        SiddhiAppRuntime siddhiAppRuntime = siddhiManager.createSiddhiAppRuntime("@App:name('TestExecutionPlan') "
+                + "define stream FooStream (symbol string, price float, volume long); " + "@info(name = 'query1') "
+                + "@sink(type='mqtt', url= 'tcp://localhost:1883', "
+                + "topic='mqtt_pause_resume', clean.session='true', message.retain='false', "
+                + "quality.of.service= '1',keep.alive= '60'," + "@map(type='xml'))"
+                + "Define stream BarStream (symbol string, price float, volume long);"
+                + "from FooStream select symbol, price, volume insert into BarStream;");
+        InputHandler fooStream = siddhiAppRuntime.getInputHandler("FooStream");
+        siddhiAppRuntime.start();
+        ArrayList<Event> arrayList = new ArrayList<Event>();
```
Comment (tishan89) [inline component/src/test/java/org/wso2/extension/siddhi/io/mqtt/source/MqttSourceTestCase.java]: declaration should be a list

File: component/src/test/java/org/wso2/extension/siddhi/io/mqtt/sink/MqttPublishTestCase.java (hunk 1 above)
Comment (HamsaV) [reply component/src/test/java/org/wso2/extension/siddhi/io/mqtt/sink/MqttPublishTestCase.java]: fixed in 51669b4

File: component/src/test/java/org/wso2/extension/siddhi/io/mqtt/sink/MqttSinkTestCase.java (hunk 3 above)
Comment (HamsaV) [reply component/src/test/java/org/wso2/extension/siddhi/io/mqtt/sink/MqttSinkTestCase.java]: fixed in 51669b4

**Pull request merged**